### ISSUE TITLE
Add service_start_time and servict_end_time to VEHICLES endpoint

### ISSUE THREAD

**Issue by Mu-yi-Zhou** [openmobilityfoundation/mobility-data-specification#899]
### Is your feature request related to a problem? Please describe.
The only vehicle info in Trips/Events/Telemetry is DEVICE_ID - we need to join them with Vehicles to get more vehicle info e.g. PROPULSION_TYPE, ACCESSIBILITY_ATTRIBUTES. There is no datetime fields currenty to indicate when a device joins the retires from the fleet, which is critical in the cases of a device gets a new VEHICLE_ID (plate number) and stays in the fleet, or a device gets upgrades to a adaptive device, which will results in two records with same DEVICE_ID. 

Joining Trips/Events/Telemetry with Vehicles on DEVICE_ID will not get us the right vehicle attributes.

### Describe the solution you'd like
Add service_start_time and servict_end_time to the VEHICLES endpoint. 

### Is this a breaking change

A breaking change would require consumers or implementors of the API to modify their code for it to continue to function (ex: renaming of a required field or the change in data type of an existing field). A non-breaking change would allow existing code to continue to function (ex: addition of an optional field or the creation of a new optional endpoint). 
 
 * No, not breaking

### Impacted Spec

For which spec is this feature being requested?

 * `provider`

**Comment by schnuerle** [comment]
Hi @jiffyclub @pierre-bouffort do you have thoughts on this being a concern of yours?

**Comment by pierre-bouffort** [comment]
Hi @schnuerle !
I think this is a valid point. We managed to hedge this risk by keeping a record of the history of the vehicle on our end, but the standard itself doesn't manage this scenario. 
So I believe the concern is legitimate.

**Comment by schnuerle** [comment]
I wonder if **service** is the right word here, which could imply maintenance. Maybe something that gets more at joining and retiring from a fleet, replacement, operation, or lifespan. `lifecycle_start` and `lifecycle_end`?

And to be clear, you are thinking of adding 2 new optional [Timestamp](https://github.com/openmobilityfoundation/mobility-data-specification/blob/main/general-information.md#timestamps) fields to this [Vehicles object](https://github.com/openmobilityfoundation/mobility-data-specification/blob/main/data-types.md#vehicles), correct?

**Comment by schnuerle** [comment]
Or is this more of an implementation issue?  

When a device/vehicle is significantly changed, then it should get a new `device_id` even if it keeps the same license plate in the `vehicle_id`. It's up to the operator to provide a new `device_id`. See [notes here](https://github.com/openmobilityfoundation/mobility-data-specification/blob/main/general-information.md#devices) saying "...device_id must remain constant for the device's lifetime of service...". 

Maybe this needs enforcement from the agency and/or clearer wording in the spec?

**Comment by mzhou0720** [comment]
> I wonder if **service** is the right word here, which could imply maintenance. Maybe something that gets more at joining and retiring from a fleet, replacement, operation, or lifespan. `lifecycle_start` and `lifecycle_end`?
> 
> And to be clear, you are thinking of adding 2 new optional [Timestamp](https://github.com/openmobilityfoundation/mobility-data-specification/blob/main/general-information.md#timestamps) fields to this [Vehicles object](https://github.com/openmobilityfoundation/mobility-data-specification/blob/main/data-types.md#vehicles), correct?

Yes that is what we are looking for. Now we are doing something similar with Pierre said, archive all the snapshots and create fields of the first and last timestamp we see the device in the vehicle endpoint.

**Comment by schnuerle** [comment]
At the public working group meeting, we discussed this at length. Here are some of the key takeaways. See [notes](https://github.com/openmobilityfoundation/mobility-data-specification/wiki/Web-conference-notes,-2025.04.10-(MDS-Working-Group)) for full details and recording.

**Key Points**

- MDS Vehicles is a real time status of vehicles, not a history
- **Vehicle property change**: when part of a vehicle is changed: accessibility, batter, tires, IoT device, license plate, color, etc (custom for each Mode). Useful for knowing use and rides with certain equipment. This is really what SFMTA wants to see (going to rename Issue/PR). Is this history a new data structure in MDS? Or a new event type in Events with descriptive text?
   - Vehicle lifecycle: when a vehicle was added to service, then finally retired. Useful for carbon footprint discussions, lifetime operations, green goals, etc. MDS Events already has decommissioned state for when its removed from service.
   - Vehicle maintenance: When vehicles have maintenance done on or off street - already covered in MDS with vehicle states in Events. Last inspection date is in Vehicles now. Some public agencies get this in monthly reports, but MDS should be able to handle this instead.
- Need to hear move from operators, vendors, data aggregators. Comment on GitHub now with challenges or issues.

**Comment by schnuerle** [comment]
There are 2 ideas in the description of this issue, vehicle 1) lifecycle and 1) property changes. Maybe also vehicle maintenance is part of this.

**Vehicle lifecycle** seemed to be less important during discussions, and is tracking of retirement is already in the MDS `decommissioned` event. Maybe there is only a need for a new event type to track when a vehicle ID is created, like `start_of_service`? That may be a breaking change. Or currently you could use when the `device_id` was first seen in the Events feed as a way to know this. Assuming operators are using `device_id` properly. There was practical and philosophical talk of how to know when to create a new device id. Maybe MDS could be update in 2.1 to clarify this. 

**Vehicle property changes** was the larger discussion and need. It's not clear how to handle this at the moment, so discussion is needed. Since this was the larger focus and is unsolved, I've renamed this issue to reflect this need. Let me know if you think that is incorrect. 

**Vendor spec compliance** is a concern where vendors are sending duplicate vehicle UUIDs in the /vehicles endpoint, which is not allowed. The spec makes this mostly clear, but the info is spread across 3 files and not always cross linked. We could make the spec clearer that there can only be 1 of a vehicle_id in the endpoint at any one time.

**Comment by gangopadhyayd** [comment]
@mzhou0720  -just a few quick clarifying questions.

a.) is the primary goal to measure each vehicle’s operational lifespan (time in service or total distance)?
b.) Assuming lifespan is key, how will you use that data—e.g. to model environmental impact per hour/km of service, evaluate depreciation, enforce maximum vehicle age limits, etc?

Or is it something else you're trying to achieve in terms of more clearly tracking a specific vehicle through its lifetime?

**Comment by Mu-yi-Zhou** [comment]
@gangopadhyayd  

our primary use case is to track vehicle property over time. As mentioned above, SFMTA created a vehicle invenotry by archiving the snapshots of vehicle endpoint. We do see reusage of device_id (vehicles of same device_id but with different attributes). Screenshot below is an example. Say when we join the trips data with the vehicle inventory (e.g. to understand trips by propulsion_type), a trip will be joined to several vehicles based on the device_id. That's why we need to know the lifecycle/in-service duration of each vehicle, that we can join the trip with vehicle based on device_id **AND** the trip_time in between the vehicle service time. Currenlty SFMTA use the data load datetime to approxiamte the vehicle lifecycle start/end time. But ideally the vendor should provide such information. 

![Image](https://github.com/user-attachments/assets/ed3fdd6e-0d3b-411d-8366-b1aa8333ac03)


Regaridn lifecycle analysis, currently the vendor send summary report to us on lifecycel, but with the start/end date, MDS can replace the existing human submission.

**Comment by schnuerle** [comment]
Based on feedback from @Mu-yi-Zhou, I updated the field names and descriptions to reflect the intention. 

Looking for feedback before merging. Is this right path to solve for this, or should instead we add a `commissioned` event type in Events, in addition to the existing `decommissioned` and `recommissioned` types?

**Comment by schnuerle** [comment]
We will be reviewing this at our next public [working group meeting](https://github.com/openmobilityfoundation/mobility-data-specification/wiki/Web-conference-notes,-2025.01.15-(MDS-Working-Group)) tomorrow. Please leave your comments now, and attend the meeting if you can.

**Comment by schnuerle** [comment]
Resolved with #931. Note we will be talking about maintenance with issue https://github.com/openmobilityfoundation/mobility-data-specification/issues/705
